### PR TITLE
task: refine DistributionTest.keepAlive

### LINKED ISSUE #26981: Workflow failure Quarkus IT - StartCommandDistTest#testWarningWhenOverridingBuildOptionsDuringStart
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

Reference GHA run: https://github.com/keycloak/keycloak/actions/runs/7848451978/job/21419988110

New unreported flaky tests, probably due to inability to start JGroups Channel due to unavailable ports.

```
2024-02-09 19:41:58,884 ERROR [org.infinispan.CONFIG] (keycloak-cache-init) ISPN000660: DefaultCacheManager start failed, stopping any running components: org.infinispan.commons.CacheException: Unable to start JGroups Channel
-- 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.startJGroupsChannelIfNeeded(JGroupsTransport.java:604)
-- 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.start(JGroupsTransport.java:477)
-- 	at org.infinispan.remoting.transport.jgroups.CorePackageImpl$2.start(CorePackageImpl.java:63)
-- 	at org.infinispan.remoting.transport.jgroups.CorePackageImpl$2.start(CorePackageImpl.java:49)
-- 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.invokeStart(BasicComponentRegistryImpl.java:616)
-- 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:607)
-- 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:576)
-- 	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:807)
-- 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startDependencies(BasicComponentRegistryImpl.java:634)
-- 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:598)
-- 	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:576)
-- 	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:807)
-- 	at org.infinispan.factories.GlobalComponentRegistry.preStart(GlobalComponentRegistry.java:284)
-- 	at org.infinispan.factories.AbstractComponentRegistry.start(AbstractComponentRegistry.java:250)
-- 	at org.infinispan.manager.DefaultCacheManager.internalStart(DefaultCacheManager.java:779)
-- 	at org.infinispan.manager.DefaultCacheManager.start(DefaultCacheManager.java:747)
-- 	at org.infinispan.manager.DefaultCacheManager.<init>(DefaultCacheManager.java:411)
-- 	at org.keycloak.quarkus.runtime.storage.legacy.infinispan.CacheManagerFactory.startCacheManager(CacheManagerFactory.java:104)
-- 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
-- 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
-- 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
-- 	at java.base/java.lang.Thread.run(Thread.java:840)
-- Caused by: java.lang.IllegalStateException: fv-az1258-409-39315: failed to find an available port in ports [49664, 49665, 49666, 49667]
-- 	at org.jgroups.protocols.FD_SOCK2.createServer(FD_SOCK2.java:326)
-- 	at org.jgroups.protocols.FD_SOCK2.start(FD_SOCK2.java:161)
-- 	at org.jgroups.stack.ProtocolStack.startStack(ProtocolStack.java:905)
-- 	at org.jgroups.JChannel.startStack(JChannel.java:922)
-- 	at org.jgroups.JChannel._preConnect(JChannel.java:800)
-- 	at org.jgroups.JChannel.connect(JChannel.java:323)
-- 	at org.jgroups.JChannel.connect(JChannel.java:317)
-- 	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.startJGroupsChannelIfNeeded(JGroupsTransport.java:602)
-- 	... 21 more
```

### Version

latest

### Expected behavior

.

### Actual behavior

.

### How to Reproduce?

.

### Anything else?

.

### PR DESCRIPTION
**Pull request by shawkins**
closes: #26981

I initially misread the intent of keepAlive, these changes should make sure that doesn't happen for others and addresses a couple of corner cases.

### CODE REVIEW

File: quarkus/tests/junit5/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java [hunk 1]
```diff
@@ -177,12 +177,11 @@ private void onKeepServerAlive(KeepServerAlive annotation) {
     public void afterEach(ExtensionContext context) throws Exception {
         DistributionTest distConfig = getDistributionConfig(context);
 
-        if (distConfig != null) {
-            if (distConfig.keepAlive()) {
-                dist.stop();
-            }
+        if (dist != null) {
+            onKeepServerAlive(context.getRequiredTestMethod().getAnnotation(KeepServerAlive.class), false);
+            dist.stop();
```
Comment (vmuzikar) [inline quarkus/tests/junit5/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java:182]: Technically, we might be stopping already stopped server here. But that should be ok.

**Review by vmuzikar — APPROVED** [review]
LGTM, thanks @shawkins!

File: quarkus/tests/junit5/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java (hunk 1 above)
Comment (shawkins) [reply quarkus/tests/junit5/src/main/java/org/keycloak/it/junit5/extension/CLITestExtension.java:182]: Right I see the close operations as being tolerant to multiple close calls.
